### PR TITLE
Schematic schematic

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    // The chain rule allows us to chain multiple rules and apply them one after the other.
+    return chain([
+        (_tree: Tree, context: SchematicContext) => {
+            // Show the options for this Schematics.
+            context.logger.info('My Full Schematic: ' + JSON.stringify(options));
+        },
+        // The schematic Rule calls the schematic from the same collection, with the options
+        // passed in. Please note that if the schematic has a schema, the options will be
+        // validated and could throw, e.g. if a required option is missing.
+        schematic('my-other-schematics', { option: true }),
```
Comment (Brocco) [inline packages/schematics/schematics/schematic/files/src/my-full-schematic/index.ts]: `my-other-schematics` should be `my-other-schematic` to match the name in `collection.json`

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+// SchematicTestRunner needs an absolute path to the collection to test.
+const collectionPath = path.join(__dirname, '../collection.json');
+
+
+describe('my-full-schematics', () => {
+    it('requires required option', () => {
+        // We test that
+        const runner = new SchematicTestRunner('schematics', collectionPath);
+        expect(() => runner.runSchematic('my-full-schematics', {}, Tree.empty())).toThrow();
```
Comment (Brocco) [inline packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts]: `my-full-schematics` --> `my-full-schematic`

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    });
+
+    it('works', () => {
+        const runner = new SchematicTestRunner('schematics', collectionPath);
+        const tree = runner.runSchematic('my-full-schematics', { required: 'str' }, Tree.empty());
```
Comment (Brocco) [inline packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts]: `my-full-schematics` --> `my-full-schematic`

File: packages/schematics/schematics/schematic/files/src/my-other-schematic/index_spec.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import * as path from 'path';
+
+
+const collectionPath = path.join(__dirname, '../collection.json');
+
+
+describe('my-other-schematics', () => {
+    it('works', () => {
+        const runner = new SchematicTestRunner('schematics', collectionPath);
+        const tree = runner.runSchematic('my-other-schematics', {}, Tree.empty());
```
Comment (Brocco) [inline packages/schematics/schematics/schematic/files/src/my-other-schematic/index_spec.ts]: `my-full-schematics` --> `my-full-schematic`

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index.ts (hunk 1 above)
Comment (hansl) [reply packages/schematics/schematics/schematic/files/src/my-full-schematic/index.ts]: Done.

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts (hunk 2 above)
Comment (hansl) [reply packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts]: Done

File: packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts (hunk 3 above)
Comment (hansl) [reply packages/schematics/schematics/schematic/files/src/my-full-schematic/index_spec.ts]: Done.

File: packages/schematics/schematics/schematic/files/src/my-other-schematic/index_spec.ts (hunk 4 above)
Comment (hansl) [reply packages/schematics/schematics/schematic/files/src/my-other-schematic/index_spec.ts]: Done.

**Pull request merged**